testing/fixtures/fixtures.go

```diff
@@ -73,3 +73,36 @@ func Clone(baseDir string, projects []Project) error {
 
 	return nil
 }
+
+// ProjectInitializerFunction defines how a single project should be initialized *after* it has already been cloned
+type ProjectInitializerFunction func(proj Project, projectDir string) error
+
+// InitializeProjects initializes provided projects in parallel using the provided initialization function. Errors are returned as a map of the Project.Name to the error thrown
+func InitializeProjects(testDir string, projects []Project, projectInitializer ProjectInitializerFunction) map[string]error {
+	var waitGroup sync.WaitGroup
+	waitGroup.Add(len(projects))
+
+	threadsafeErrorsByProject := threadableErrorsByProject{
+		errorsByProject: make(map[string]error, len(projects)),
+	}
+
+	for _, project := range projects {
+		go func(proj Project) {
+			defer waitGroup.Done()
+			err := projectInitializer(proj, filepath.Join(testDir, proj.Name))
+
+			// separate the initialization step from locking to decrease time locking out other writers
+			threadsafeErrorsByProject.Lock()
+			threadsafeErrorsByProject.errorsByProject[proj.Name] = err
```

Comment:
Why not panic immediately like we do in `Clone`? I think we should do one or the other in both cases.

Personally, my preference is for an immediate panic. Yes, there is value in seeing all the errors, but I don't think the added value is worth the extra code.

Reply:
for cases like `npm` where there can be some failures that we may want to handwave away, it will be useful to inspect the results and change what happens depending on it. The current implementation ignores npm errors, as an example as to where this can be used

Reply:
Also, if cloning does not work, we have bigger issues that will likely need to be addressed ASAP, so panicking there seems more fitting to terminate the program as quickly as possible

Reply:
looking back, I think you're right